tests/kernel_bundle/kernel_bundle_spec_const.cpp

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "kernel_bundle.h"
+
+namespace kernel_bundle_spec_const {
+
+constexpr sycl::specialization_id<int> SpecName(5);
+constexpr sycl::specialization_id<int> OtherSpecName(5);
+
+TEST_CASE(
+    "Check specialization constants functionality with empty kernel_bundle",
+    "[kernel_bundle]") {
+  sycl::context ctx;
+  const auto always_false_selector = [](auto device_image) { return false; };
```

Comment:
It's a bit strange that device image selectors work differently from device selectors, but that's a spec issue, the test code here is correct.